### PR TITLE
Tiny associated type inference fixes

### PR DESCRIPTION
**Pull request by slavapestov**
- Regression test for https://bugs.swift.org/browse/SR-11392
- Fixes https://bugs.swift.org/browse/SR-11407 / rdar://problem/54979757

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: stdlib/public/core/NativeSet.swift [hunk 1]
```diff
@@ -282,6 +282,9 @@ extension _NativeSet {
 }
 
 extension _NativeSet: _SetBuffer {
+  @usableFromInline
```
Comment (slavapestov) [inline stdlib/public/core/NativeSet.swift]: This goes away in the next PR.

**Review by DougGregor — APPROVED** [review]
Thanks!

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Pull request merged**